### PR TITLE
Set resource version for SPOD update

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Every update would fail because the resource version won't be set. This patch fixes this corner case.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Fixed SPOD update bug for default profiles where resource version is not set.
```

### CODE REVIEW

File: internal/pkg/manager/spod/spod_controller.go [hunk 1]
```diff
@@ -281,6 +281,9 @@ func (r *ReconcileSPOd) handleUpdate(
 		foundProfile := &sccmpv1alpha1.SeccompProfile{}
 		var err error
 		if err = r.client.Get(ctx, pKey, foundProfile); err == nil {
```
Comment (hasheddan) [inline internal/pkg/manager/spod/spod_controller.go:283]: Could consider using an `Applicator` :) https://github.com/crossplane/crossplane-runtime/blob/53874539b31013368170eb1090d2f083745d5051/pkg/resource/api.go#L168

**Comment by saschagrunert** [comment]
Whoop this already has been fixed :shrug:
